### PR TITLE
kernel: hil: sensors: AirQualityDriver improvements

### PR DESCRIPTION
**Pull request by alistair23**
### Pull Request Overview

A few small improvements to the `AirQualityDriver` before implementations are added

### Documentation Updated

- [X] Updated the relevant files in `/docs`, or no updates are required.

### Formatting

- [X] Ran `make prepush`.

### CODE REVIEW

File: kernel/src/hil/sensors.rs [hunk 1]
```diff
@@ -36,6 +36,19 @@ pub trait AirQualityDriver<'a> {
...
+    /// quality.
+    ///
+    /// This is an optional call and doesn't have to be used, but on most
+    /// hardware can be used to improve the measurement accuracy.
+    ///
+    /// This function might return the following errors:
+    /// - `BUSY`: Indicates that the hardware is busy with an existing
+    ///           operation or initilisation/calibration.
+    /// - `NOSUPPORT`: Indicates that this data type isn't supported.
+    fn specify_enviroment(&self, temp: Option<u32>, humidity: Option<u32>)
```
Comment (hudson-ayers) [inline kernel/src/hil/sensors.rs:49]: We should document the intended units for temp and humidity here

**Review by hudson-ayers — APPROVED** [review]

**Review by bradjc — APPROVED** [review]

File: kernel/src/hil/sensors.rs [hunk 2]
```diff
@@ -36,13 +36,29 @@ pub trait AirQualityDriver<'a> {
     /// Set the client to be notified when the capsule has data ready.
     fn set_client(&self, client: &'a dyn AirQualityClient);
 
+    /// Specify the temperature and humidity used in calculating the air
+    /// quality.
+    ///
+    /// The temperature is specified in degress Celsius and the humidity
```
Comment (bradjc) [inline kernel/src/hil/sensors.rs:42]:
```suggestion
    /// The temperature is specified in degrees Celsius and the humidity
```

File: kernel/src/hil/sensors.rs [hunk 3]
```diff
@@ -36,13 +36,29 @@ pub trait AirQualityDriver<'a> {
...
+    /// is specified as a percentage.
+    ///
+    /// This is an optional call and doesn't have to be used, but on most
+    /// hardware can be used to improve the measurement accuracy.
+    ///
+    /// This function might return the following errors:
+    /// - `BUSY`: Indicates that the hardware is busy with an existing
+    ///           operation or initialisation/calibration.
+    /// - `NOSUPPORT`: Indicates that this data type isn't supported.
+    fn specify_enviroment(&self, temp: Option<u32>, humidity: Option<u32>)
```
Comment (brghena) [inline kernel/src/hil/sensors.rs:52]: Per the #2965 discussion, we should probably use `i32` for temperature here, since it can be less than zero Celsius.

**Review by hudson-ayers — APPROVED** [review]

**Pull request merged**